test/mpi/dtpools/src/dtpools_internal.c

```diff
@@ -532,6 +532,9 @@ int DTPI_Struct_create(struct DTPI_Par *par, DTP_t dtp)
     }
 
     /* cleanup buffers */
+    if (basic_type_displs) {
```

Comment:
Actually, `free(3)` accepts `NULL` pointer (see the manpage), so you don't need this guard.

Reply:
This needs to be addressed.

Reply:
Now this commit adds the logic to free the previously unfreed buffer using the same approach I was using before for other buffers. In the next commit I remove all the `if`s from every buffer that is initialized to `NULL`.